python/ray/serve/tests/unit/test_deployment_state.py

```diff
@@ -367,6 +337,8 @@ def mock_deployment_state() -> Tuple[DeploymentState, Mock, Mock]:
         "ray.serve._private.deployment_state.ActorReplicaWrapper",
         new=MockReplicaActorWrapper,
     ), patch("time.time", new=timer.time), patch(
+        "ray.serve._private.deployment_scheduler.get_head_node_id",
+    ) as mock_get_head_node_id, patch(
```

Comment:
No need for this mock, let's just pass the head node ID into the replica scheduler constructor (the function is only called once in its constructor)